htslib/kstring.h

```diff
@@ -140,8 +140,8 @@ static inline int ks_resize(kstring_t *s, size_t size)
 		char *tmp;
 		kroundup_size_t(size);
 		tmp = (char*)realloc(s->s, size);
-		if (!tmp)
-			return -1;
+		if (!tmp && size)
```

Comment:
Actually, this can't happen if input `size` is 0, because `s->m` and `size` are both unsigned so `s->m < size` will always be false.  However it can if `size` is greater than `SIZE_MAX / 2` as `kroundup_size_t(size)` then overflows and sets `size` to zero.  So this should actually be an out-of-memory error, given that you can never allocate a buffer that big.

This kroundup overflow is likely to be a problem elsewhere too.